general/design-dev-cycle.md

```diff
@@ -0,0 +1,19 @@
+Product/Design development cycle for source{d} application:
```

Comment:
this should be also linked from the **engineering** section

Reply:
I think this can be done later, after the content of this doc is polished and there is a consensus from all sides.